src/openrct2-ui/drawing/engines/opengl/TextureCache.cpp

```diff
@@ -128,36 +87,61 @@ void TextureCache::CreateAtlasesTexture()
         glGetIntegerv(GL_MAX_ARRAY_TEXTURE_LAYERS, &_atlasesTextureIndicesLimit);
         if (_atlasesTextureDimensions < _atlasesTextureIndicesLimit) _atlasesTextureIndicesLimit = _atlasesTextureDimensions;
 
-        AllocateAtlasesTexture();
-
-        _atlasesTextureInitialised = true;
+        glGenTextures(1, &_atlasesTexture);
+        glBindTexture(GL_TEXTURE_2D_ARRAY, _atlasesTexture);
+        glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+        glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+        glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+        
+        glGenTextures(1, &_paletteTexture);
+        glBindTexture(GL_TEXTURE_RECTANGLE, _paletteTexture);
+        glTexParameteri(GL_TEXTURE_RECTANGLE, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+        glTexParameteri(GL_TEXTURE_RECTANGLE, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+        glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+        GeneratePaletteTexture();
+
+        _initialized = true;
         _atlasesTextureIndices = 0;
     }
 }
 
-void TextureCache::AllocateAtlasesTexture()
+void TextureCache::GeneratePaletteTexture()
 {
-    // Create an array texture to hold all of the atlases
-    glGenTextures(1, &_atlasesTexture);
-    glBindTexture(GL_TEXTURE_2D_ARRAY, _atlasesTexture);
-    glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-    glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
-    glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+    rct_drawpixelinfo dpi = CreateDPI(256, PALETTE_TO_G1_OFFSET_COUNT + 5);
```

Comment:
Is the +5 because of water?

Reply:
It's +1 for the no-op palette, and +(5 - 1) for the water palette.